File: ros2_example_bazel_installed/test/shimmed_sentinel.cc
#include <cstdlib>
#include <iostream>

const char * kShimmedSentinel = "_BAZEL_ROS2_RULES_SHIMMED";

int main(int argc, char* argv[]) {
  bool shimmed = (nullptr != getenv(kShimmedSentinel));
  bool app_present = (nullptr != getenv("AMENT_PREFIX_PATH"));

  std::cout << "{\"shimmed\": " << shimmed
    << ", \"AMENT_PREFIX_PATH present\": " << app_present << "}";
  return 0;
}

File: ros2_example_bazel_installed/test/shimmed_sentinel.py
import json
import os


assert __name__ == '__main__'

result = {
    'shimmed': '_BAZEL_ROS2_RULES_SHIMMED' in os.environ,
    'AMENT_PREFIX_PATH present': 'AMENT_PREFIX_PATH' in os.environ
}

print(json.dumps(result))
